Lib/test/test_peepholer.py

```diff
@@ -421,6 +421,14 @@ def f():
             return 0
         self.assertEqual(f(), 1)
 
+    def test_if_with_if_expression(self):
+        # Check bpo-37289
+        def f():
+            if True if f.__code__ else None:
```

Comment:
The truthiness of `f.__code__` is unrelated to this issue. I suggest to use:
```python
def f(x):
    if (True if x else False):
        return True
    return False
self.assertTrue(f(True))
```

Reply:
Done in 860c7c6.

I wanted to have exactly the same bytecode (including attribute access) but as the important part is the JUMP_FORWARD followed by LOAD_CONST and POP_JUMP_IF_FALSE I agree that your proposal is better :)